src/hybrid/parse_hybrid.c

```diff
@@ -466,6 +474,60 @@ static void copyRequestConfig(RequestConfig *dest, const RequestConfig *src) {
   dest->BM25STD_TanhFactor = src->BM25STD_TanhFactor;
 }
 
+// Helper function to get LIMIT value from parsed aggregation pipeline
+static size_t getLimitFromPipeline(Pipeline *pipeline) {
+  RS_ASSERT(pipeline);
+
+  PLN_ArrangeStep *arrangeStep = AGPLN_GetArrangeStep(&pipeline->ap);
+  if (arrangeStep && arrangeStep->isLimited && arrangeStep->limit > 0) {
+    return (size_t)arrangeStep->limit;
+  }
+  return 0;
+}
+
+// Helper function to check if LIMIT was explicitly provided
+static bool hasExplicitLimitInPipeline(Pipeline *pipeline) {
+  if (!pipeline) return false;
+
+  PLN_ArrangeStep *arrangeStep = AGPLN_GetArrangeStep(&pipeline->ap);
+  return (arrangeStep && arrangeStep->isLimited);
+}
+
+/**
+ * Apply LIMIT parameter fallback logic to KNN K and WINDOW parameters.
+ *
+ * When LIMIT is explicitly provided but KNN K or WINDOW are not explicitly set,
+ * this function applies the LIMIT value as a fallback for those parameters instead of their
+ * defaults (unless they have been explicitly set).
+ * This ensures consistent behavior where LIMIT acts as a unified size hint
+ * for hybrid search operations.
+ *
+ * @param mergePipeline The pipeline to extract LIMIT from
+ * @param vectorRequest The vector request containing KNN parameters
+ * @param hybridParams The hybrid parameters containing WINDOW settings
+ */
+static void applyLimitParameterFallbacks(Pipeline *mergePipeline,
+                                       AREQ *vectorRequest,
+                                       HybridPipelineParams *hybridParams) {
+  size_t limitValue = getLimitFromPipeline(mergePipeline);
+  bool hasExplicitLimit = hasExplicitLimitInPipeline(mergePipeline);
```

Comment:
Will this not get the default value of `10`, since the KNN step sets the `isLimited` field as well?